### PR TITLE
Basic style guide

### LINKED ISSUE #1: Style Guide
Add style guide to .github directory

What to cover in guide
- [x] Indentation
- [x] Trailing Whitespace
- [x] Variable declarations
- [x] Naming 
- [x] Comments
- [x] If-else nesting
- [x] Function size

### PR DESCRIPTION
**Pull request by Munsterberg**
closes #1

### CODE REVIEW

File: .github/STYLE-GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Style Guide for Adeo
+
+A short but sweet style guide for the Adeo project. Please follow all rules in
+this guide as closely as possible.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: No need for a new line.

File: .github/STYLE-GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## Table of Contents
+
+### Formatting
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Use heading 4 instead of 3.

File: .github/STYLE-GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+* [Indentation](#indentation)
+* [Whitespace](#whitespace)
+* [Variable Declarations](#variable-declarations)
+* [Line Limit](#line-limit)
+* [Braces](#braces)
+* [Comments](#comments)
+* [Naming](#naming)
+* [Functions](#functions)
+
+### ES6
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+In progress
+
+## Formatting
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+### Indentation
+Use 2 spaces for indentation. Please do not use tabs, spaces only.
+
+### Whitespace
+Before creating a PR please double check to make sure there is no trailing
+whitespace in your commits.
+
+### Variable Declarations
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+* [Comments](#comments)
+* [Naming](#naming)
+* [Functions](#functions)
+
+### ES6
+In progress
+
+## Formatting
+
+### Indentation
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+Wrong
+```javascript
+var foo = 'bar',
+    numbers = [1, 2, 3],
+    emptyObj = {};      
+```
+
+### Line Limit
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+Limit your lines to 80 characters. If using something like Atom or Sublime
+search for a package that allows for this functionality.
+
+### Braces
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+Wrong
+```javascript
+function sample()
+{
+  console.log('wrong');
+}
+```
+
+### Comments
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+Use single line comments always. Do not use multiline comments.
+
+### Naming
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+lowerCamelCase will be used for variables and functions. CamelCase will be used
+for Classes.
+
+### Functions
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

File: .github/STYLE-GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+Use 2 spaces for indentation. Please do not use tabs, spaces only.
+
+### Whitespace
+Before creating a PR please double check to make sure there is no trailing
+whitespace in your commits.
+
+### Variable Declarations
+We limit you to one variable declaration per line.
+
+Correct
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: ~~Correct~~ **Correct**

File: .github/STYLE-GUIDE.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+### Line Limit
+Limit your lines to 80 characters. If using something like Atom or Sublime
+search for a package that allows for this functionality.
+
+### Braces
+Have opening braces on the same line as your statement.
+
+Correct
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: ~~Correct~~ **Correct**

File: .github/STYLE-GUIDE.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+```javascript
+function sample() {
+  console.log('correct');
+}
+```
+
+Wrong
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: ~~Wrong~~ **Wrong**

File: .github/STYLE-GUIDE.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+We limit you to one variable declaration per line.
+
+Correct
+```javascript
+var foo = 'bar';
+var numbers = [1, 2, 3];
+var emptyObj = {};
+```
+
+Wrong
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: ~~Wrong~~ **Wrong**

File: .github/STYLE-GUIDE.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+
+### ES6
+In progress
+
+## Formatting
+
+### Indentation
+Use 2 spaces for indentation. Please do not use tabs, spaces only.
+
+### Whitespace
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: See [above](https://github.com/extramundane/adeo/pull/4/files#r64993556)

**Comment by atjonathan** [comment]
@Munsterberg Maybe have the **correct** after **wrong**?

**Comment by atjonathan** [comment]
### **LGTM :+1:**

Just make those changes and we should be good for merging!

**Comment by Munsterberg** [comment]
I added a link to the Airbnb guide, which has a ton of information and best practices. If anyone has trouble understanding etc. just refer them there.

File: .github/STYLE-GUIDE.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
+# Style Guide for Adeo
+A short but sweet style guide for the Adeo project. Please follow all rules in
+this guide as closely as possible.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+## Table of Contents
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+* [Indentation](#indentation)
+* [Whitespace](#whitespace)
+* [Variable Declarations](#variable-declarations)
+* [Line Limit](#line-limit)
+* [Braces](#braces)
+* [Comments](#comments)
+* [Naming](#naming)
+* [Functions](#functions)
+
+#### ES6
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+For an up to date extensive style guide for ES6 please refer to [airbnb's javascript style guide.](https://github.com/airbnb/javascript)
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+### Formatting
+#### Indentation
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+* [Variable Declarations](#variable-declarations)
+* [Line Limit](#line-limit)
+* [Braces](#braces)
+* [Comments](#comments)
+* [Naming](#naming)
+* [Functions](#functions)
+
+#### ES6
+For an up to date extensive style guide for ES6 please refer to [airbnb's javascript style guide.](https://github.com/airbnb/javascript)
+### Formatting
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+#### Indentation
+Use 2 spaces for indentation. Please do not use tabs, spaces only.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+#### Whitespace
+Before creating a PR please double check to make sure there is no trailing
+whitespace in your commits.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+#### Variable Declarations
+We limit you to one variable declaration per line.
+
+**Correct**
+```javascript
+const foo = 'bar';
+const numbers = [1, 2, 3];
+const emptyObj = {};
+```
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add **wrong** before **right**

File: .github/STYLE-GUIDE.md [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+**Wrong**
+```javascript
+const foo = 'bar',
+    numbers = [1, 2, 3],
+    emptyObj = {};      
+```
+#### Line Limit
+Limit your lines to 80 characters. If using something like Atom or Sublime
+search for a package that allows for this functionality.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+const foo = 'bar';
+const numbers = [1, 2, 3];
+const emptyObj = {};
+```
+**Wrong**
+```javascript
+const foo = 'bar',
+    numbers = [1, 2, 3],
+    emptyObj = {};      
+```
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+search for a package that allows for this functionality.
+#### Braces
+Have opening braces on the same line as your statement.
+
+**Correct**
+```javascript
+function sample() {
+  console.log('correct');
+}
+```
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add **wrong** before **right**

File: .github/STYLE-GUIDE.md [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+**Wrong**
+```javascript
+function sample()
+{
+  console.log('wrong');
+}
+```
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+#### Comments
+Use single line comments always. Do not use multiline comments.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

File: .github/STYLE-GUIDE.md [hunk 31]
```diff
@@ -0,0 +1,64 @@
...
+#### Naming
+lowerCamelCase will be used for variables and functions. CamelCase will be used
+for Classes.
```
Comment (atjonathan) [inline .github/STYLE-GUIDE.md]: Add line

**Pull request merged**